libs/algo/odc/algo/_dask.py

```diff
@@ -108,6 +119,81 @@ def unpack_chunksize(chunk: int, N: int) -> Tuple[int, ...]:
     return tuple(chunk for _ in range(nb)) + (last_chunk,)
 
 
+def _roi_from_chunks(chunks: Tuple[int, ...]) -> Iterator[slice]:
+    off = 0
+    for v in chunks:
+        off_next = off + v
+        yield slice(off, off_next)
+        off = off_next
+
+
+def _split_chunks(chunks: Tuple[int, ...], max_chunk: int) -> Iterator[Tuple[int, int, slice]]:
+    """
+    For every input chunk split it into smaller chunks.
+    Return a list of tuples describing output chunks and their relation to input chunks.
+
+    Output: [(dst_idx: int, src_idx: int, src_roi: slice]
+
+    Note that every output chunk has only one chunk on input,
+    so chunking might be irregular on output. This is by design, to avoid
+    creating cross chunk dependencies.
+    """
+    dst_idx = 0
+    for src_idx, src_sz in enumerate(chunks):
```

Comment:
Sometimes these abbreviations make it harder to read. I think this one is ok, `src_sz` says 'source size' to me. Just wanted to comment, though.

Reply:
I disagree, I find long variable names are much much harder to pattern match visually.
I find that understanding what variable IS in a given context is a "once off operation" human brain does, while seeing where it is being used is done over and over again. If you had `source_index` and `source_size` and then `destination_index` and `destination_size`  then

(a) things do not align vertically
```
source_size
destination_size
# vs
dst_sz
src_sz
```
(b) Lines become much longer making them harder to grasp at a glance, and you are also more likely to end up splitting them vertically, making comprehension even harder still.

That's why it is idiomatic in some domains (like computer vision) to shorten prefixes for source and destination variables to `src_{whatever}, dst_{whatever}`.